app/js/components/tabs/tabcomponents/conceptComponent.jsx

```diff
@@ -1,35 +1,150 @@
 import React, {Component} from 'react';
 
 class ConceptComponent extends Component {
-    componentDidMount(){}
+    constructor(props) {
+        super(props);
+        this.state = {
+            conceptsResults: [],
+            searchResults: [],
+            verbose: false
+        };
+        this.allConcepts = [];
+        this.searchConcept = this.searchConcept.bind(this);
+        this.checkVerbose = this.checkVerbose.bind(this);
+        this.loadConcepts = this.loadConcepts.bind(this);
+    }
+    /**
+     * Check if verbose is clicked and set verbose state to true
+     * @param {*} event 
+     */
+    checkVerbose(event) {
+        const verbose = document.getElementById('verbose').checked;
+        this.setState({
+            verbose: verbose
+        })
+    }
+    /**
+     * Load concepts suggestions from the backend based on entered input
+     * entered keyword
+     * @param {*} event 
+     */
+    loadConcepts(event) {
+            const conceptName = event.target.value,
+                regex = new RegExp(conceptName, 'gi')
+            if(conceptName.length > 2){
+                this.allConcepts = [];
+                new Promise((resolve, reject) => {
+                    this.props.fetchData('/concept?v=full').then(data => {
```

Comment:
Not really, it is searching the concepts and loading only the results that match the specified search input as specified on line 41 & 50.